route/route.go

```diff
@@ -947,6 +978,46 @@ func (r *Router) Watch(req *grpc_health_v1.HealthCheckRequest, server grpc_healt
 	})
 }
 
+// startGRPCHealthMonitor starts a goroutine that periodically checks the health of the system and updates the grpc health server
+func (r *Router) startGRPCHealthMonitor() {
+	const (
+		system      = "" // empty string represents the generic health of the whole system (corresponds to "ready")
+		systemReady = "ready"
+		systemAlive = "alive"
+	)
+	r.iopLogger.Debug().Logf("running grpc health monitor")
+
+	setStatus := func(svc string, stat bool) {
+		if stat {
+			r.hsrv.SetServingStatus(svc, grpc_health_v1.HealthCheckResponse_SERVING)
+		} else {
+			r.hsrv.SetServingStatus(svc, grpc_health_v1.HealthCheckResponse_NOT_SERVING)
+		}
+	}
+
+	r.doneWG.Add(1)
+	go func() {
+		defer r.doneWG.Done()
+		// TODO: Does this time need to be configurable?
+		watchticker := time.NewTicker(3 * time.Second)
```

Comment:
I wonder if we could make this configurable so that it can be set based on the probe in k8s https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

Reply:
It could definitely be configurable, but the k8s timing is unrelated. The longest you can go without detecting a failure is the sum of this time plus the k8s time. Is there a real case for ever trying to be significantly faster than this? Also, it's rare enough that there's no real benefit in going much slower. To me, it seems like the practical range is from 1-10 seconds, and 3 is a good compromise.

Reply:
I'm going to merge this and we can add configuration later if we decide we want it.